core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -3460,22 +3465,31 @@ class ViewMapping<
 
     m_impl_handle = handle_type(reinterpret_cast<pointer_type>(record->data()));
 
+    functor_type functor =
+        execution_space_specified
+            ? functor_type(exec_space, (value_type*)m_impl_handle,
+                           m_impl_offset.span(), alloc_name)
+            : functor_type((value_type*)m_impl_handle, m_impl_offset.span(),
+                           alloc_name);
+
+    if (false) {
```

Comment:
Please guard that code so it is only there when absolutely necessary.
This will be helpful if one tries to revisit later.

Reply:
I need the code outside the `if constexpr` block so that I don't run into the `cudaErrorInvalidDeviceFunction` error in the new test.

Reply:
I meant macro guards